control/controller.go

```diff
@@ -399,7 +405,7 @@ func (q *Query) State() State {
 // transitionTo will transition from one state to another. If a list of current states
 // is given, then the query must be in one of those states for the transition to succeed.
 // This method must be called with a lock and it must be called from within the run loop.
-func (q *Query) transitionTo(newState State, currentState ...State) bool {
+func (q *Query) transitionTo(newState State, currentState ...State) (context.Context, bool) {
```

Comment:
It's not intuitive why you would both set `q.currentCtx` and return it here, so it would be good to add a comment to this effect.

Reply:
I guess you're right that's not really needed anymore other than winking at the developer to create a race condition.